### PR TITLE
LIBRARY OVERHAUL

### PR DESCRIPTION
**Pull request by N3X15**
NEEDS MAP CHANGES, WAITING FOR OTHER MAP CHANGES FIRST.
# Changes
- Library computers are now actual computers.
- Lookups from MySQL are paginated (no more 25,000-record results)
- Manuals use the same code as regular books
- Code for programmatic content books
- Books that use a wiki page massively simplified and centralized
- Books written by users now have a ckey associated with them so Pomf can yell at Hime for uploading 50 fucking copies of furry trap porn
# TODO
- May go under VGWS:
  - [x] Admin panel for deleting books
  - [ ] Script for deleting books with broken links
- [x] Markup support for books
- [x] Map changes due to repathed computers

### CODE REVIEW

**Comment by NigglyWiggly** [comment]
>Library computers are now actual computers.

Nexis I think I love you

**Comment by Duny-** [comment]
Based bluethumbsup

**Comment by IratePirate** [comment]
YES.

**Comment by NigglyWiggly** [comment]
Also, does this mean I can upgrade the library computer with parts from R&D to make it print books without the 10 second delay?

**Comment by N3X15** [comment]
@NigglyWiggly No

File: code/modules/library/computers/checkout.dm [hunk 1]
```diff
@@ -0,0 +1,366 @@
+/*
+ * Library Computer
+ *
+ * TODO: Make this an actual /obj/machinery/computer that can be crafted from circuit boards and such
+ * It is August 22nd, 2012... This TODO has already been here for months.. I wonder how long it'll last before someone does something about it.
+ * It's 25th of January in the year of (our) Lord 2015... And it's still not a computer.
```
Comment (Iamgoofball) [inline code/modules/library/computers/checkout.dm]: needs a new commemorative comment

**Comment by ZomgPonies** [comment]
This is great, can't wait to see it finished.

**Comment by N3X15** [comment]
Okay, this needs some testing and then I'll have pomf up it as part uno.

**Pull request merged**